internal/metadataproviders/system/metadata.go

```diff
@@ -128,6 +129,22 @@ func (p systemMetadataProvider) reverseLookup(ipAddresses []string) (string, err
 	return "", fmt.Errorf("reverseLookup failed to convert IP addresses to name: %w", err)
 }
 
-func (p systemMetadataProvider) HostID() (string, error) {
-	return machineid.ID()
+func (p systemMetadataProvider) HostID(ctx context.Context) (string, error) {
+	res, err := resource.New(ctx,
```

Comment:
It's a bit "funny" how you need create a resource that includes the "host.id" attribute here, only to "peel" the resource to get just the "host.id" attribute value, and in the end you put that value in an attribute of another resource. This is due to the structure of the contrib's `system.Provider` interface :shrug: